packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,253 @@
+	};
+
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Author Settings' ) }>
+					<SelectControl
+						label={ __( 'Author' ) }
+						value={ postAuthor }
+						options={ authors.map( ( { id, name } ) => {
+							return {
+								value: id,
+								label: name,
+							};
+						} ) }
+						onChange={ ( newAuthorId ) => {
+							setPostAuthor( newAuthorId );
+							getAuthorDetails( newAuthorId, setAuthorDetails );
+						} }
+					/>
+					<ToggleControl
+						label={ __( 'Show avatar' ) }
+						checked={ showAvatar }
+						onChange={ () =>
+							props.setAttributes( { showAvatar: ! showAvatar } )
+						}
+					/>
+					{ showAvatar && (
+						<SelectControl
+							label={ __( 'Avatar size' ) }
+							value={ props.attributes.avatarSize }
+							options={ avatarSizes }
+							onChange={ ( size ) => {
+								props.setAttributes( {
+									avatarSize: Number( size ),
+								} );
+							} }
+						/>
+					) }
+					<ToggleControl
+						label={ __( 'Show bio' ) }
+						checked={ showBio }
+						onChange={ () =>
+							props.setAttributes( { showBio: ! showBio } )
+						}
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Text settings' ) }>
+					<FontSizePicker
+						value={ fontSize.size }
+						onChange={ setFontSize }
+					/>
+				</PanelBody>
+			</InspectorControls>
+
+			{ InspectorControlsColorPanel }
+
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( nextAlign ) => {
+						props.setAttributes( { align: nextAlign } );
+					} }
+				/>
+				<BlockColorsStyleSelector
+					TextColor={ TextColor }
+					BackgroundColor={ BackgroundColor }
+				>
+					{ ColorPanel }
+				</BlockColorsStyleSelector>
+			</BlockControls>
+
+			<TextColor>
+				<BackgroundColor>
+					<div
+						ref={ ref }
+						className={ classnames( blockClassNames, {
+							[ `has-text-align-${ align }` ]: align,
+						} ) }
+						style={ blockInlineStyles }
+					>
+						{ showAvatar && authorDetails && (
+							<div className="wp-block-post-author__avatar">
+								<img
+									width={ avatarSize }
+									src={
+										authorDetails.avatar_urls[ avatarSize ]
+									}
+									alt={ authorDetails.name }
+								/>
+							</div>
+						) }
+						<div className="wp-block-post-author__content">
+							{ ( ! RichText.isEmpty( byline ) ||
+								isSelected ) && (
+								<RichText
+									className="wp-block-post-author__byline"
+									multiline={ false }
+									placeholder={ __( 'Write byline …' ) }
+									withoutInteractiveFormatting
+									allowedFormats={ [
+										'core/bold',
+										'core/italic',
+										'core/strikethrough',
+									] }
+									value={ byline }
+									onChange={ ( value ) =>
+										props.setAttributes( { byline: value } )
+									}
+								/>
+							) }
+							<p className="wp-block-post-author__name">
+								{ authorDetails.name }
```

Comment:
[Above](https://github.com/WordPress/gutenberg/pull/19894/files#diff-7e82f718ff736f624069097db0995cb1R179), there's a condition to check whether `authorDetails` is truthy before accessing properties on it. There's no such guards here. Should there be?

Reply:
> [Above](https://github.com/WordPress/gutenberg/pull/19894/files#diff-7e82f718ff736f624069097db0995cb1R179), there's a condition to check whether `authorDetails` is truthy before accessing properties on it. There's no such guards here. Should there be?

This conditional has been incorporated into the proposed changes at #22359.